kernel/src/sched/priority_scheduler.rs

```diff
@@ -268,7 +269,7 @@ trait PreemptSchedInfo {
 
     fn priority(&self) -> Priority;
 
-    fn cpu_affinity(&self) -> SpinLockGuard<CpuSet, PreemptDisabled>;
+    fn cpu_affinity(&self) -> CpuSet;
```

Comment:
Returning an owned `CpuSet` means we will use the affinity at this time to do CPU selection. I wonder if this will lead to a kind of TOCTOU bug, or there are actually no users of `sched_setaffinity` who set others' affinity?

For the `priority` method, my concern is the same. Maybe we should never allow this kind of early fetching of the inner value of an atomic field here?

Reply:
Maybe this TOCTOU is kind of a innocuous behavior since `sched_setaffinity` (if indeed implemented the right behavior in the future) will initiate a migration. My concern is that a lock may lead to performance degradation, since the scheduler will acquire additional locks during selection.

Reply:
You are right. If `sched_setaffinity` can trigger migration, there won't be any problem. I'm now with the atomic version.